src/rdf4cpp/rdf/Literal.cpp

```diff
@@ -0,0 +1,56 @@
+#include "Literal.hpp"
+
+#include <rdf4cpp/rdf/IRI.hpp>
+#include <rdf4cpp/rdf/storage/node/LiteralBackend.hpp>
+
+namespace rdf4cpp::rdf {
+
+Literal::Literal(const Node::NodeID &id) : Node(id) {}
+Literal::Literal() : Node() {}
+Literal::Literal(const std::string &lexical_form, Node::NodeStorage &node_storage)
+    : Node(BackendNodeHandle{node_storage.get_string_literal(lexical_form).second}) {}
+Literal::Literal(const std::string &lexical_form, const IRI &datatype, Node::NodeStorage &node_storage)
+    : Node(BackendNodeHandle{
+              node_storage.get_typed_literal(
+                                  lexical_form,
+                                  datatype.handle_.id())
+                      .second}) {}
+Literal::Literal(const std::string &lexical_form, const std::string &lang, Node::NodeStorage &node_storage)
+    : Node(BackendNodeHandle{node_storage.get_lang_literal(lexical_form, lang).second}) {}
+
+
+IRI Literal::datatype() const {
```

Comment:
IRI is good for now, 
would change that in the future to an own class, which can validate the literal value.

Reply:
Yes, we will also need a concept on how exactly we integrate datatype support.